src/arch/xtensa/include/arch/string.h

```diff
@@ -32,9 +32,66 @@
 #ifndef __INCLUDE_ARCH_STRING_SOF__
 #define __INCLUDE_ARCH_STRING_SOF__
 
-void *xthal_memcpy(void *dst, const void *src, size_t len);
+#include <errno.h>
 
 #define arch_memcpy(dest, src, size) \
 	xthal_memcpy(dest, src, size)
 
+#if __XCC__ && !defined(CHECK)
+#define arch_memset(ptr, size)	\
+		memset_s(ptr, size, 0, size)
```

Comment:
Same `size` should not be used as the buffer size and set operation range. This might mask existing calls where the buffer size is not explicitly provided, and effectively disable size checks (always pass).

Reply:
Same situation  only used for bzero operations on buffers - will change naming convention.